### DISCUSSION TITLE
I made a React template for bootstrapping a PocketBase app

### DISCUSSION THREAD

**Discussion by lucafaggianelli** [pocketbase/pocketbase#5129] [Show and tell]
Hey there,

I discovered PocketBase quite recently and I find it really great for starting a side project in just few minutes!
So I looked for a React template to get started even faster, though I didn't find any... so well I thought that it could be useful to share a template with you to get started with a _full-stack_ PocketBase app, that is PB + a frontend.

Here you go:
https://github.com/lucafaggianelli/pocket-saas

![image](https://github.com/pocketbase/pocketbase/assets/4471043/530aa3a4-0bdc-4353-ab6b-297d7d9ab9b6)

![image](https://github.com/pocketbase/pocketbase/assets/4471043/741e9695-5be0-4fd9-8209-c08b38ec89b9)

The template mostly includes PocketBase authentication and it's the first rough release, but I'm adding other features as soon as I develop them for my app.

👉 🧨 Let me know if you find it useful and what would you add as feats

## 🥞 Tech stack
- [React](https://reactjs.org/)
- [Tailwind CSS](https://tailwindcss.com/)
- [shadcn/ui](https://ui.shadcn.com/)
- [Tanstack Router](https://tanstack.com/router/latest/docs/framework/react/overview)
- [TypeScript](https://www.typescriptlang.org/)
- [Vite](https://vitejs.dev/)
- [pnpm](https://pnpm.io/)

## 🛣️ Roadmap
- [ ] 3rd party integrations: allow to connect apps via OAuth2 (not user auth), this includes creating a PB Collection and several routes
- [ ] Stripe payments: add a PB Collection and webhook routes for Stripe
- [ ] Forms for record creation: add some example how to create records from a form via JS
- [ ] CI/CD & deployment: add a Github action to automatically build and deploy on the cloud

**Comment by osaxon** [comment]
Hey, thanks for sharing this. I found your post whilst looking for Pocketbase + React resources and looks like we're working with the same stack pretty much.
How are you handling refreshing auth tokens? I've opted for a 'usePocketbase' hook that I call in the root App component of my project. I then pass down the token, user and the pb client to React router as context.

Here's my repo if you want to take a look. the custom hook is in src/context/usePocketbase.ts
https://github.com/osaxon/vite-pocketbase

**Comment by lucafaggianelli** [comment]
hey, indeed I didn't think about it, but yeah your solution looks the most appropriate, I'll add auth refresh as well in my template, thanks for pointing it out!

And yes it really seems we're using the same tech stack :)
